File: Source/Alternet.UI.Pal/SizerFlags.cpp
#include "SizerFlags.h"

namespace Alternet::UI
{
    SizerFlags::SizerFlags()
    {
    }

    SizerFlags::~SizerFlags()
    {
    }

    void* SizerFlags::CreateSizerFlags(int proportion)
    {
        return new wxSizerFlags(proportion);
    }
    
    int SizerFlags::GetDefaultBorder() 
    {
        return wxSizerFlags::GetDefaultBorder();
    }
    
    float SizerFlags::GetDefaultBorderFractional() 
    {
        return wxSizerFlags::GetDefaultBorderFractional();
    }
    
    int SizerFlags::GetProportion(void* handle) 
    {
        return ((wxSizerFlags*)handle)->GetProportion();
    }
    
    int SizerFlags::GetFlags(void* handle) 
    {
        return ((wxSizerFlags*)handle)->GetFlags();
    }
    
    int SizerFlags::GetBorderInPixels(void* handle) 
    {
        return ((wxSizerFlags*)handle)->GetBorderInPixels();
    }
    
    void SizerFlags::Proportion(void* handle, int proportion) 
    {
        ((wxSizerFlags*)handle)->Proportion(proportion);
    }
    
    void SizerFlags::Expand(void* handle) 
    {
        ((wxSizerFlags*)handle)->Expand();
    }
    
    void SizerFlags::Align(void* handle, int alignment) 
    {
        ((wxSizerFlags*)handle)->Align(alignment);
    }
    
    void SizerFlags::Center(void* handle) 
    {
        ((wxSizerFlags*)handle)->Center();
    }
    
    void SizerFlags::CenterVertical(void* handle) 
    {
        ((wxSizerFlags*)handle)->CenterVertical();
    }
    
    void SizerFlags::CenterHorizontal(void* handle) 
    {
        ((wxSizerFlags*)handle)->CenterHorizontal();
    }
    
    void SizerFlags::Top(void* handle) 
    {
        ((wxSizerFlags*)handle)->Top();
    }
    
    void SizerFlags::Left(void* handle) 
    {
        ((wxSizerFlags*)handle)->Left();
    }
    
    void SizerFlags::Right(void* handle) 
    {
        ((wxSizerFlags*)handle)->Right();
    }
    
    void SizerFlags::Bottom(void* handle) 
    {
        ((wxSizerFlags*)handle)->Bottom();
    }
    
    void SizerFlags::Border(void* handle, int direction, int borderInPixels) 
    {
        ((wxSizerFlags*)handle)->Border(direction, borderInPixels);
    }
    
    void SizerFlags::Border2(void* handle, int direction) 
    {
        ((wxSizerFlags*)handle)->Border(direction);
    }
    
    void SizerFlags::DoubleBorder(void* handle, int direction) 
    {
        ((wxSizerFlags*)handle)->DoubleBorder(direction);
    }
    
    void SizerFlags::TripleBorder(void* handle, int direction) 
    {
        ((wxSizerFlags*)handle)->TripleBorder(direction);
    }
    
    void SizerFlags::HorzBorder(void* handle) 
    {
        ((wxSizerFlags*)handle)->HorzBorder();
    }
    
    void SizerFlags::DoubleHorzBorder(void* handle) 
    {
        ((wxSizerFlags*)handle)->DoubleHorzBorder();
    }
    
    void SizerFlags::Shaped(void* handle) 
    {
        ((wxSizerFlags*)handle)->Shaped();
    }
    
    void SizerFlags::FixedMinSize(void* handle) 
    {
        ((wxSizerFlags*)handle)->FixedMinSize();
    }
    
    void SizerFlags::ReserveSpaceEvenIfHidden(void* handle) 
    {
        ((wxSizerFlags*)handle)->ReserveSpaceEvenIfHidden();
    }
}

File: Source/Alternet.UI/WxWidgets/Sizer/SizerFlags.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Alternet.UI
{
    // https://docs.wxwidgets.org/3.2/classwx_sizer_flags.html
    internal class SizerFlags : DisposableObject<IntPtr>, ISizerFlags
    {
        internal SizerFlags(int proportion = 0)
            : base(Native.SizerFlags.CreateSizerFlags(proportion), true)
        {
        }

        public static int GetDefaultBorder()
        {
            return Native.SizerFlags.GetDefaultBorder();
        }

        public static float GetDefaultBorderFractional()
        {
            return Native.SizerFlags.GetDefaultBorderFractional();
        }

        int ISizerFlags.GetDefaultBorder()
        {
            return Native.SizerFlags.GetDefaultBorder();
        }

        float ISizerFlags.GetDefaultBorderFractional()
        {
            return Native.SizerFlags.GetDefaultBorderFractional();
        }

        public int GetProportion()
        {
            return Native.SizerFlags.GetProportion(Handle);
        }

        public SizerFlag GetFlags()
        {
            return (SizerFlag)Native.SizerFlags.GetFlags(Handle);
        }

        public int GetBorderInPixels()
        {
            return Native.SizerFlags.GetBorderInPixels(Handle);
        }

        public ISizerFlags Proportion(int proportion)
        {
            Native.SizerFlags.Proportion(Handle, proportion);
            return this;
        }

        public ISizerFlags Expand()
        {
            Native.SizerFlags.Expand(Handle);
            return this;
        }

        public ISizerFlags Align(GenericAlignment alignment)
        {
            Native.SizerFlags.Align(Handle, (int)alignment);
            return this;
        }

        public ISizerFlags Center()
        {
            Native.SizerFlags.Center(Handle);
            return this;
        }

        public ISizerFlags CenterVertical()
        {
            Native.SizerFlags.CenterVertical(Handle);
            return this;
        }

        public ISizerFlags CenterHorizontal()
        {
            Native.SizerFlags.CenterHorizontal(Handle);
            return this;
        }

        public ISizerFlags Top()
        {
            Native.SizerFlags.Top(Handle);
            return this;
        }

        public ISizerFlags Left()
        {
            Native.SizerFlags.Left(Handle);
            return this;
        }

        public ISizerFlags Right()
        {
            Native.SizerFlags.Right(Handle);
            return this;
        }

        public ISizerFlags Bottom()
        {
            Native.SizerFlags.Bottom(Handle);
            return this;
        }

        public ISizerFlags Border(GenericDirection direction, int borderInPixels)
        {
            Native.SizerFlags.Border(Handle, (int)direction, borderInPixels);
            return this;
        }

        public ISizerFlags Border(GenericDirection direction = GenericDirection.All)
        {
            Native.SizerFlags.Border2(Handle, (int)direction);
            return this;
        }

        public ISizerFlags DoubleBorder(GenericDirection direction = GenericDirection.All)
        {
            Native.SizerFlags.DoubleBorder(Handle, (int)direction);
            return this;
        }

        public ISizerFlags TripleBorder(GenericDirection direction = GenericDirection.All)
        {
            Native.SizerFlags.TripleBorder(Handle, (int)direction);
            return this;
        }

        public ISizerFlags HorzBorder()
        {
            Native.SizerFlags.HorzBorder(Handle);
            return this;
        }

        public ISizerFlags DoubleHorzBorder()
        {
            Native.SizerFlags.DoubleHorzBorder(Handle);
            return this;
        }

        public ISizerFlags Shaped()
        {
            Native.SizerFlags.Shaped(Handle);
            return this;
        }

        public ISizerFlags FixedMinSize()
        {
            Native.SizerFlags.FixedMinSize(Handle);
            return this;
        }

        // makes the item ignore window's visibility status
        public ISizerFlags ReserveSpaceEvenIfHidden()
        {
            Native.SizerFlags.ReserveSpaceEvenIfHidden(Handle);
            return this;
        }
    }
}